tests/e2e-otel/k8sleaderelectorext/check_leader_election.sh

```
@@ -0,0 +1,49 @@
+FOUND_METRICS=false
+FOUND_LEADER_LOG=false
+
+while ! $FOUND_LEASE || ! $FOUND_METRICS || ! $FOUND_LEADER_LOG; do
+    # Check that the Lease object exists
+    if ! $FOUND_LEASE && kubectl -n $NAMESPACE get lease $LEASE_NAME > /dev/null 2>&1; then
+        HOLDER=$(kubectl -n $NAMESPACE get lease $LEASE_NAME -o jsonpath='{.spec.holderIdentity}')
+        echo "Lease '$LEASE_NAME' found with holder: $HOLDER"
+        FOUND_LEASE=true
+    fi
+
+    PODS=($(kubectl -n $NAMESPACE get pods -l $LABEL_SELECTOR -o jsonpath='{.items[*].metadata.name}'))
```

Shellcheck (0.11.0)

[warning] Prefer mapfile or read -a to split command output (or quote to avoid splitting).

(SC2207)